services/api/src/utils/schema.js

```diff
@@ -471,6 +471,7 @@ function flattenQuery(query, schema, root = {}, rootPath = []) {
   for (let [key, value] of Object.entries(query)) {
     if (key.includes('.')) {
       // Custom dot syntax is allowed and is already flattened, so skip.
+      root[key] = value;
```

Comment:
I don't think this should be needed? The flattenQuery method transforms the query only if it needs it so anything `.find()` would understand shouldn't need to be touched?

Reply:
then the problem might be else where 

currently the query like this doesnt work
```
Schema.search({
   "nested.field": "search"
})
```

Reply:
awesome let me confirm ... but yeah looks like it has been fixed

Reply:
The test is bad ... the query that it triggers is `{}`

Reply:
sorry not sure what you mean? so the test should be `Model.search({})`?